Gemfile

```diff
@@ -0,0 +1,56 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '3.0.2'
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails', branch: 'main'
+gem 'rails', '~> 6.1.4', '>= 6.1.4.1'
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3', '~> 1.4'
+# Use Puma as the app server
+gem 'puma', '~> 5.0'
+# Use SCSS for stylesheets
+gem 'sass-rails', '>= 6'
+# Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
+gem 'webpacker', '~> 5.0'
+# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
+gem 'turbolinks', '~> 5'
+# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
+gem 'jbuilder', '~> 2.7'
+# Use Redis adapter to run Action Cable in production
+# gem 'redis', '~> 4.0'
+# Use Active Model has_secure_password
+# gem 'bcrypt', '~> 3.1.7'
+
+# Use Active Storage variant
+# gem 'image_processing', '~> 1.2'
+
+# Reduces boot times through caching; required in config/boot.rb
+gem 'bootsnap', '>= 1.4.4', require: false
+
+group :development, :test do
```

Comment:
What are `group` ?

Reply:
`group` allow you to specify groups of dependencies, where you can perform operations on the entire group. For instance, you can install all gems except certain specify groups. You can instruct bundler to skip the development group by `bundle install --without development` during installation.

Reply:
ok. Then, what are the kind of groups there are and can you set your own group?

Reply:
Groups are user defined. Common ones seem to be based on the environments like `test`, `development` & `production`.

You can set up your own group as such:
```
group :group-name do
  gem 'gem-1'
  gem 'gem-2'
end
```

Reply:
Ok. So, if say I define:

```
group :abc do
  gem 'gem-1'
end
```

How do I get gem-1 to be installed only?

Reply:
By default, the gems outside of named groups are in the `:default` group. Assuming that the defaults are empty here, we could run `bundle install --with abc`. Alternatively, `bundle install --standalone abc` should work as well.